### ISSUE TITLE
Relative routes to assets don't work

### ISSUE THREAD

**Issue by tiptronic** [timberio/gitdocs#31]
### Prenote: I don't use a docs.json for my tests.

In my docs I have a common 'assets' directory, containing all my imagery.

assets
   imageA.png
fileA.md
folderA
    fileB.md

To link imageA.png from fileA.md I use:

``` ![](assets/imageA.png)```

To link imageA from fileB.md I use:
``` ![](../assets/imageA.png)```

Both fail to link and both files are linked against:
```/assets/imageA.png```

You can also check this in your docs at markdown.md#links

**Comment by zsherman** [comment]
Ah we need to figure out how to support this, currently I believe gitdocs looks for images in a `docs/public` folder, cc @etimberg.

**Comment by tiptronic** [comment]
If it looks in a public folder that would work for me, although it then isn't proper markdown, imo... so the better choice would be to start using relative paths.

**Comment by zsherman** [comment]
Agreed, we'll work on this asap.

**Comment by tiptronic** [comment]
I'm sorry to demand stuff - I'm used to help, but my knowledge about React is close to zero 😔
If I can help testing (or whatever) please let me know. I like this project pretty much!

**Comment by etimberg** [comment]
Thinking about a solution to this, what if we did:

Copy all static images files (.png. .jpg, etc) found in `/docs/` that are not in `/docs/public/` with the same relative folder structure. Ie, copy `/docs/assets/1.png` to `/output/assets/1.png`.

Doing this would require changing the react-static `onStart` and `onBuild`  [hooks](https://github.com/timberio/gitdocs/blob/master/static.config.js#L175-L185)

A downside of this approach is the possibility that multiple files copy over themselves in the output directory. Suppose we have `/docs/public/assets/1.png` and `/docs/assets/1.png`, the user should receive some kind of error as both files would appear in the same place `/docs-dist/assets/1.png`.

**Comment by tiptronic** [comment]
tbh - I don't think this is an ideal solution. What's the problem keeping the assets where they are and symlinking the folder... (sorry, if this is complete bs... but I haven't looked into the code, so I don't know what's really going on there)
But -fwiw- it seems gitbooks handles that properly

**Comment by etimberg** [comment]
I don't think we can symlink the folders because I think there is a requirement that `gitdocs build` produces something that can copied and deployed so I think we'd have to copy. @zsherman definitely has more insight though.
